cdc/processor/pipeline/sink.go

```diff
@@ -351,15 +352,26 @@ func (n *sinkNode) HandleMessage(ctx context.Context, msg pipeline.Message) (boo
 			}
 		}
 	case pipeline.MessageTypeBarrier:
-		n.barrierTs = msg.BarrierTs
-		if err := n.flushSink(ctx, n.resolvedTs); err != nil {
+		if err := n.UpdateBarrierTs(ctx, msg.BarrierTs); err != nil {
 			return false, errors.Trace(err)
 		}
 	}
 	return true, nil
 }
 
+func (n *sinkNode) UpdateBarrierTs(ctx context.Context, ts model.Ts) error {
+	atomic.StoreUint64(&n.barrierTs, ts)
+	if err := n.flushSink(ctx, n.resolvedTs); err != nil {
+		return errors.Trace(err)
+	}
+	return nil
+}
+
 func (n *sinkNode) Destroy(ctx pipeline.NodeContext) error {
+	return n.ReleaseResource(ctx)
+}
+
+func (n *sinkNode) ReleaseResource(ctx context.Context) error {
```

Comment:
It is better to change this method to a private method due to it is only called by `tableActor`.

Reply:
done